14 card component for roadmap items

MiaTexnes opened the pull request:
Added the roadmap section with a timeline showing 6 phases. Each phase has a card with an icon, title, description, and status (done, in progress, to do).
Built the RoadmapCard component and RoadmapTimeline. Added dark mode support and basic accessibility.
I may have gone a bit overboard, so some related issues may have been completed as part of this.

MiaTexnes committed:
feat(roadmap): add timeline with status-aware cards

- Reusable card component accepting props (icon, title, text, status)
- Phase icons and timeline visualization
- Dark mode support with CSS variables
- Refactored to use constants and path aliases

MiaTexnes committed:
feat: add timeline with cards

- Reusable card component accepting props (icon, title, text, status)
- Phase icons and timeline visualization
- Dark mode support with CSS variables

MiaTexnes committed:
refactor: clean up comments and improve code readability in RoadmapCard component

MiaTexnes committed:
fix(translations): correct spelling errors in roadmap descriptions

Anclagen committed:
fix: dark-light toggle streamlining and fixing

Anclagen committed:
fix: removed App.css and move relevant styles to index.css

Anclagen committed:
chore: spacing guidelines draft

MiaTexnes committed:
refactor(styles): migrate to Tailwind theme variables and add dark mode support

- Remove custom App.css and integrate styles into index.css
- Add Tailwind theme with color and typography variables
- Implement automatic dark mode toggling
- Add spacing guidelines documentation for consistency

docs: add spacing guidelines for layout and components

MiaTexnes committed:
refactor(roadmap): use Tailwind v4 auto-toggle color variables

Anclagen commented on src/features/roadmap/components/RoadmapTimeline.jsx:
```diff
@@ -1,35 +1,85 @@
+import searchIcon from "@/assets/text-search.svg";
+import settingsIcon from "@/assets/cog.svg";
+import fileUserIcon from "@/assets/file-user.svg";
+import mountainIcon from "@/assets/mountain-snow.svg";
```
The icons the design team are using come from Lucide which has a react package we can use. 
```jsx
import { Users, Rocket, TextSearch, Cog, FileUser, MountainSnow } from 'lucide-react';
```

Anclagen replied:
If the dev branch is merge in the lucid install should already be sorted so its just a case of importing and using the relevant icons.

Anclagen reviewed (changes requested):
Looks good, just about ready to merge.

I see on the project board the task for installing Lucid for icons hasn't been completed. Can you install Lucide React, and update the icons to use this package in this branch, then this can be merged.

https://lucide.dev/guide/packages/lucide-react
You can search for icons here too:
https://lucide.dev/icons

MiaTexnes committed:
refactor(roadmap): replace custom SVG icons with lucide-react components

- Add lucide-react dependency to package.json
- Delete unused SVG assets (cog, file-user, mountain-snow, rocket, text-search, users)
- Update RoadmapCard and RoadmapTimeline to import and use Lucide React icons
- Change icon prop to Icon component for dynamic rendering
- Improve icon styling and accessibility with component-based approach

Anclagen reviewed (dismissed):
Nice!
